https://gitlab.jlse.anl.gov/intel-dga/simulation/kokkos/-/jobs/153552 says something like
```
[...]]/atomic_ref.hpp:125:3: error: static assertion failed due to requirement 'detail::integral_constant<bool, false>::value': Invalid default memory_order for atomics.  Valid defaults are: relaxed, acq_rel, seq_cst
  static_assert(
  ^
[...]/atomic_ref.hpp:269:14: note: in instantiation of template class 'sycl::detail::atomic_ref_base<unsigned int, sycl::memory_order::acquire, sycl::memory_scope::device, sycl::access::address_space::generic_space>' requested here
    : public atomic_ref_base<T, DefaultOrder, DefaultScope, AddressSpace> {
             ^
[...]/atomic_ref.hpp:660:27: note: in instantiation of template class 'sycl::detail::atomic_ref_impl<unsigned int, sycl::memory_order::acquire, sycl::memory_scope::device, sycl::access::address_space::generic_space>' requested here
class atomic_ref : public detail::atomic_ref_impl<T, DefaultOrder, DefaultScope,
                          ^
[...]/tpls/desul/include/desul/atomics/Compare_Exchange_SYCL.hpp:32:59: note: in instantiation of template class 'sycl::atomic_ref<unsigned int, sycl::memory_order::acquire, sycl::memory_scope::device, sycl::access::address_space::generic_space>' requested here
  sycl_atomic_ref<unsigned int, MemoryOrder, MemoryScope> dest_ref(
                                                          ^
[...]/tpls/desul/include/desul/atomics/Generic.hpp:37:20: note: in instantiation of function template specialization 'desul::Impl::device_atomic_compare_exchange<int, desul::MemoryOrderAcquire, desul::MemoryScopeDevice>' requested here
      return Impl::device_atomic_compare_exchange(dest, cmp, val, order, scope);)
                   ^
[...]/kokkos/sycl-release-develop/core/src/OpenMP/Kokkos_OpenMP_Instance.cpp:38:22: note: in instantiation of function template specialization 'desul::atomic_compare_exchange<int, desul::MemoryOrderAcquire, desul::MemoryScopeDevice>' requested here
  while (1 == desul::atomic_compare_exchange(&m_pool_mutex, 0, 1,
```

--- a/tpls/desul/include/desul/atomics/Adapt_SYCL.hpp
+++ b/tpls/desul/include/desul/atomics/Adapt_SYCL.hpp
@@ -38,6 +38,11 @@ using sycl_memory_scope = std::conditional_t<extended_namespace,
                                              sycl::memory_scope>;
 
 //<editor-fold desc="SYCL memory order">
+// The default memory order for sycl::atomic_ref
+// can be seq_cst, acq_rel, or relaxed according to the
+// "SYCL 2020 Specification (revision 6)", see
+// https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:atomic-references.
+// Thus, we map MemoryOrderAcquire and MemoryOrderRelease to acq_rel.
 template <class MemoryOrder, bool extended_namespace = true>
 struct SYCLMemoryOrder;
 
@@ -49,12 +54,12 @@ struct SYCLMemoryOrder<MemoryOrderSeqCst, extended_namespace> {
 template <bool extended_namespace>
 struct SYCLMemoryOrder<MemoryOrderAcquire, extended_namespace> {
   static constexpr sycl_memory_order<extended_namespace> value =
-      sycl_memory_order<extended_namespace>::acquire;
+      sycl_memory_order<extended_namespace>::acq_rel;
 };
 template <bool extended_namespace>
 struct SYCLMemoryOrder<MemoryOrderRelease, extended_namespace> {
   static constexpr sycl_memory_order<extended_namespace> value =
-      sycl_memory_order<extended_namespace>::release;
+      sycl_memory_order<extended_namespace>::acq_rel;
 };
 template <bool extended_namespace>
 struct SYCLMemoryOrder<MemoryOrderAcqRel, extended_namespace> {

